src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -592,7 +592,12 @@ internal void ProcessAuthentication(LazyAsyncResult lazyResult)
                 //  A trick to discover and avoid cached sessions.
                 _CachedSession = CachedSessionStatus.Unknown;
 
-                ForceAuthentication(Context.IsServer, null, asyncRequest);
+                if (asyncRequest != null)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Cancellation exceptions shouldn't be thrown out of the invocations synchronously; they should be passed out in a task.  That's going to require more refactoring, e.g. to do what @Drawaes has been doing and converting things from being Task-over-APM to instead be APM-over-Task, at which point implementing the cancellation support in that way will be trivial.